packages/aws/src/lib/CostAndUsageReports.ts

```diff
@@ -527,30 +516,37 @@ export default class CostAndUsageReports {
     start: Date,
     end: Date,
     grouping: GroupBy,
+    tagNames: string[],
```

Comment:
Sure. Let me know if this is clear and then I'll add it in:

> AWS' Cost and Usage Reporting (CUR) translates tags' names to names that are valid Athena column names. On top of this, it also adds a prefix to distinguish between user-created tags and AWS-internal tags.
> This isn't documented anywhere, but the behaviour appears to be that a tag such as `SourceRepository` will be `user:SourceRepository` in CUR, and `resource_tags_user_source_repository` in Athena. (AWS-internal tags will be prefixed with `aws:` instead of `user:` in CUR.)

We then need to add these column names to the query (so that Athena returns their values) and to the `GROUP BY` clause (so that the values are aggregated by the tag values).

Looking at this again with fresh eye, I'm not sure why I continued with using indices (like the existing code) rather than names for the column names in the `GROUP BY` - I'll try swapping over to column names to make it clearer.

The rest of the changes are formatting changes or variable extraction to make the query easier to read (all the extra linebreaks created by Prettier made it difficult to follow, at least for me).

Reply:
`GROUP BY` now uses column names rather than indices: https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/pull/904/commits/026636b7cf203305ee6ce333a3d4828f6614a50c

Reply:
Comment has been added in [`1c66593`](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/pull/904/commits/1c6659328c923487338cd66708696ddb41837eab).